net/portmapper/upnp.go

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package portmapper
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/tailscale/goupnp/dcps/internetgateway2"
+	"golang.org/x/sync/errgroup"
+	"inet.af/netaddr"
+)
+
+// DisableUPnP indicates whether to attempt UPnP mapping.
```

Comment:
does this need to be exported?

Reply:
I was thinking that we want to be able to shut it down from Control if UPnP explodes. Currently it's not used elsewhere though

Reply:
let's unexported for now.

we should definitely have a safety knob to let us disable this later but we should make it match.

see tailcfg.go's type Debug and:

```go
                        setControlAtomic(&controlUseDERPRoute, resp.Debug.DERPRoute)
                        setControlAtomic(&controlTrimWGConfig, resp.Debug.TrimWGConfig)
```